faq.md

```diff
@@ -575,6 +575,144 @@ If you are using the Flutter plugin for IntelliJ, you can launch the app in
+frequently used to process lists of values in the framework.
+
+* Event-driven programming: User interactions are represented by event objects
+that are dispatched to callbacks registered with event handlers. Screen updates
+are triggered by a similar callback mechanism. The
+[Listenable](https://docs.flutter.io/flutter/foundation/Listenable-class.html)
+class, which is used as the basis of the animation system, formalises a
+subscription model for events with multiple listeners.
+
+* Class-based object-orientated programming: Most of the APIs of the framework
+are built using classes with inheritance. We use an approach whereby we define
+very high-level APIs in our base classes, then specialise them iteratively in
+subclasses. For example, our render objects have a base class
+([RenderObject](https://docs.flutter.io/flutter/rendering/RenderObject-class.html))
+that is agnostic regarding the coordinate system, and then we have a subclass
+([RenderBox](https://docs.flutter.io/flutter/rendering/RenderBox-class.html))
+that introduces the opinion that the geometry should be based on the Cartesian
+coordinate system (x/width and y/height).
+
+* Prototype-based object-orientated programming: The
+[ScrollPhysics](https://docs.flutter.io/flutter/widgets/ScrollPhysics-class.html)
+class chains instances to compose the physics that apply to scrolling
+dynamically at runtime. This lets the system compose, for example, paging
+physics with platform-specific physics, without the platform having to be
+selected at compile time.
+
+* Imperative programming: Straightforward imperative programming, usually
+paired with state encapsulated within an object, is used where it provides the
+most intuitive solution. For example, tests are written in an imperative style,
+first describing the situation under test, then listing the invariants that the
+test must match, then advancing the clock or inserting events as necessary for
+the test.
+
+* Reactive programming: The widget and element trees are sometimes described as
+reactive, because new inputs provided in a widget's constructor are immediately
+propagated as changes to lower-level widgets by the widget's build method, and
+changes made in the lower widgets (e.g. in response to user input) propagate
+back up the tree via event handlers. Aspects of both functional-reactive and
+imperative-reactive are present in the framework, depending on the needs of the
+widgets. For example, the material
+[Divider](https://docs.flutter.io/flutter/material/Divider-class.html) class has
+a build method that consists of just an expression describing how this widget
+reacts to changes in its configuration (a functional reactive widget), while the
+[Chip](https://docs.flutter.io/flutter/material/Chip-class.html) class has a
+build method that constructs a list of children over several statements,
+describing how this widget reacts to changes in its configuration (an imperative
+reactive widget).
```

Comment:
Again, move the parenthetical remark to right after "Chip class"

Reply:
same here, this is trying to distinguish "an expression describing how this widget reacts to changes in its configuration" (the earlier text) from "a build method that constructs a list of children over several statements, describing how this widget reacts to changes in its configuration".